### ISSUE TITLE
Allow to configure functional swagger option parameters and plugins

### ISSUE THREAD

**Issue by nexinto-gmbh** [scttcper/koa2-swagger-ui#40]
Hello,

currently its necessary to configure requestInterceptor and also a plugin.
So I extends a local copy of your project to add the new functionality.

If you can adopt my changes to your code, I can switch back to your project.

Thx

```js
//index.js
...
module.exports = function koaSwagger(config) {
...
  // added two helper to check and render functions directly
  Handlebars.registerHelper('strfnc', fnc => fnc)
  Handlebars.registerHelper('isset', function (conditional, options) { return !!conditional ? options.fn(this) : options.inverse(this) })
...
 ```

```hbs
{{! index.hbs }}
...
  window.onload = function() {
    const options = {{{json swaggerOptions}}};
    {{! Check if is set and set the four functional options }}
    {{#isset swaggerOptions.requestInterceptor}}
      options.requestInterceptor = {{{strfnc swaggerOptions.requestInterceptor}}}
    {{/isset}}
    {{#isset swaggerOptions.responseInterceptor}}
    options.responseInterceptor = {{{strfnc swaggerOptions.responseInterceptor}}}
    {{/isset}}
    {{#isset swaggerOptions.responseInterceptor}}
    options.modelPropertyMacro = {{{strfnc swaggerOptions.modelPropertyMacro}}}
    {{/isset}}
    {{#isset swaggerOptions.parameterMacro}}
    options.parameterMacro = {{{strfnc swaggerOptions.parameterMacro}}}
    {{/isset}}


    {{! extends presets and plugins - both must be an array or undefined }}
    options.presets = [
      SwaggerUIBundle.presets.apis,
      // yay ES6 modules ↘
      Array.isArray(SwaggerUIStandalonePreset) ? SwaggerUIStandalonePreset : SwaggerUIStandalonePreset.default,
      {{{swaggerOptions.presets}}}
    ];
    options.plugins = [
      SwaggerUIBundle.plugins.DownloadUrl,
      {{#if hideTopbar}}
        HideTopbarPlugin
      {{/if}},
      {{{swaggerOptions.plugins}}}
    ];
...
```

**Comment by scttcper** [comment]
Can you do a pr please?

**Comment by nexinto-gmbh** [comment]
https://github.com/scttcper/koa2-swagger-ui/pull/41
Done

**Comment by scttcper** [comment]
published as 2.7.0
